lms/static/scripts/frontend_apps/components/AutoGradingConfigurator.tsx

```diff
@@ -0,0 +1,208 @@
+   * - cumulative: both annotations and replies will be counted together for
+   *               the grade calculation
+   * - separately: students will have different annotation and reply goals.
+   */
+  activityCalculation: 'cumulative' | 'separately';
+
+  /**
+   * Required number of annotations if activityCalculation is 'separately' or
+   * annotations+replies otherwise
+   */
+  requiredAnnotations: number;
+
+  /**
+   * Required number of replies if activityCalculation is 'separately'
+   */
+  requiredReplies?: number;
+};
+
+type AnnotationsGoalInputGroupProps = {
+  children?: ComponentChildren;
+  gradingType: GradingType;
+  value: number;
+  onChange: (newValue: number) => void;
+};
+
+/**
+ * Controls containing a number input to set the amount of required annotations
+ * or replies
+ */
+function AnnotationsGoalInputGroup({
+  children,
+  gradingType,
+  value,
+  onChange,
+}: AnnotationsGoalInputGroupProps) {
+  const inputId = useId();
+
+  return (
+    <div className="flex gap-2 items-center">
+      <label
+        className="grow flex justify-between items-center"
+        htmlFor={inputId}
+      >
+        {children}
+        <span className="uppercase font-semibold">
+          {gradingType === 'all_or_nothing' ? 'Minimum' : 'Goal'}
+        </span>
+      </label>
+      <Input
+        id={inputId}
+        classes="max-w-14"
+        type="number"
+        required
+        min={1}
+        step={1}
+        value={value}
+        onChange={e => onChange(Number((e.target as HTMLInputElement).value))}
+      />
+    </div>
+  );
+}
+
+export type AutoGradingConfiguratorProps = {
+  autoGradingConfig: AutoGradingConfig;
+  updateAutoGradingConfig: (newConfig: AutoGradingConfig) => void;
+};
+
+/**
+ * Allows instructors to enable auto grading for an assignment, and provide the
+ * configuration to determine how to calculate every student's grade.
+ */
+export default function AutoGradingConfigurator({
+  autoGradingConfig,
+  updateAutoGradingConfig,
+}: AutoGradingConfiguratorProps) {
+  const {
+    active = false,
+    gradingType,
+    activityCalculation,
+    requiredAnnotations,
+    requiredReplies = 1,
+  } = autoGradingConfig;
+  const updatePartialConfig = useCallback(
+    (partialConfig: Partial<AutoGradingConfig>) =>
+      updateAutoGradingConfig({ ...autoGradingConfig, ...partialConfig }),
+    [autoGradingConfig, updateAutoGradingConfig],
+  );
+
+  const gradingTypeId = useId();
+  const activityCalculationId = useId();
+
+  return (
+    <div className="flex flex-col gap-y-3">
+      <Checkbox
+        checked={active}
+        checkedIcon={CheckboxCheckedFilledIcon}
+        onChange={e =>
+          updatePartialConfig({
+            active: (e.target as HTMLInputElement).checked,
+          })
+        }
+      >
+        Enabled automatic participation grading
+      </Checkbox>
+      {active && (
+        <>
+          <div>
+            <p id={gradingTypeId} className="font-semibold mb-1">
```

Comment:
A heading would probably be more appropriate than a paragraph here, although I suspect screen readers will ignore the type given the element's role as a label for a radio group.

Reply:
What would be the right heading type here? I'm never too sure when it's not a very obvious h1 or h2. Maybe h3?

Reply:
The card heading ("Assignment details") is an h1. The section headings on the left should probably be headings as well, but currently they are not, so that would be h2. In that case this would be h3.